### PR TITLE
*: add a flag EncodeType for response to use Arrow/Default format to decode.

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve? 

Now the `SelectResponse` use two field to save Arrow/Default format data. But they are both bytes array and can use the same field.  We can add field `EncodeType` instead of field `RowBatchData`.

### What is changed and how it works?

Remove the field `RowBatchData` and add the flag field `EncodeType`.

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

**Comment by XuHuaiyu** [comment]
Why the build failed?

File: distsql/select_result.go [hunk 1]
```diff
@@ -147,18 +147,15 @@ func (r *selectResult) NextRaw(ctx context.Context) (data []byte, err error) {
 func (r *selectResult) Next(ctx context.Context, chk *chunk.Chunk) error {
 	chk.Reset()
 	// Check the returned data is default/arrow format.
-	if r.selectResp == nil || (len(r.selectResp.RowBatchData) == 0 && r.respChkIdx == len(r.selectResp.Chunks)) {
+	if r.selectResp == nil || r.respChkIdx == len(r.selectResp.Chunks) {
 		err := r.getSelectResp()
 		if err != nil || r.selectResp == nil {
 			return err
 		}
 		// TODO(Shenghui Wu): add metrics
```
Comment (XuHuaiyu) [inline distsql/select_result.go]: Should this be moved to line 157

**Comment by wshwsh12** [comment]
I change the proto and adjust tidb and tikv's code. 
`integration-compatibility-test` will use both tidb-master and tidb-pr/12536(this pr) to run test. And one of them will fail.
If the two related pr merge, the test will pass.

File: distsql/select_result.go (hunk 1 above)
Comment (wshwsh12) [reply distsql/select_result.go]: ok.

**Pull request merged**